feature/preview/src/main/java/com/google/jetpackcamera/feature/preview/PreviewViewModel.kt

```diff
@@ -136,6 +158,77 @@ class PreviewViewModel @AssistedInject constructor(
         }
     }
 
+    /**
+     * Returns the difference between two [CameraAppSettings] as a queue of
+     * [Pair]<[SettingNames], [Any]>.
+     */
+    private fun getSettingsDiff(
+        oldCameraAppSettings: CameraAppSettings,
+        newCameraAppSettings: CameraAppSettings
+    ): ArrayDeque<Pair<KProperty<Any?>, Any?>> {
+        val queue = ArrayDeque<Pair<KProperty<Any?>, Any?>>()
+        CameraAppSettings::class.memberProperties.forEach { property ->
+            if (property.get(oldCameraAppSettings) != property.get(newCameraAppSettings)) {
+                queue.add(Pair(property, property.get(newCameraAppSettings)))
+            }
+        }
+        return queue
+    }
+
+    /**
+     * Iterates through a queue of [Pair]<[KProperty], [Any]> and attempt to apply them to
+     * [CameraUseCase].
+     */
+    private suspend fun applySettingsDiff(
+        newSettingsDiff: ArrayDeque<Pair<KProperty<Any?>, Any?>>
+    ) {
+        newSettingsDiff.forEach { pair ->
+            when (pair.first) {
+                CameraAppSettings::cameraLensFacing -> {
+                    cameraUseCase.setLensFacing(pair.second as LensFacing)
+                }
+
+                CameraAppSettings::flashMode -> {
+                    cameraUseCase.setFlashMode(pair.second as FlashMode)
+                }
+
+                CameraAppSettings::captureMode -> {
+                    cameraUseCase.setCaptureMode(pair.second as CaptureMode)
+                }
+
+                CameraAppSettings::aspectRatio -> {
+                    cameraUseCase.setAspectRatio(pair.second as AspectRatio)
+                }
+
+                CameraAppSettings::previewStabilization -> {
+                    cameraUseCase.setPreviewStabilization(pair.second as Stabilization)
+                }
+
+                CameraAppSettings::videoCaptureStabilization -> {
+                    cameraUseCase.setVideoCaptureStabilization(pair.second as Stabilization)
+                }
+
+                CameraAppSettings::targetFrameRate -> {
+                    cameraUseCase.setTargetFrameRate(pair.second as Int)
+                }
+
+                CameraAppSettings::lowLightBoost -> {
+                    cameraUseCase.setLowLightBoost(pair.second as LowLightBoost)
+                }
+
+                CameraAppSettings::audioMuted -> {
+                    cameraUseCase.setAudioMuted(pair.second as Boolean)
+                }
+
+                CameraAppSettings::imageFormat -> {
+                    cameraUseCase.setImageFormat(pair.second as ImageOutputFormat)
+                }
```

Comment:
None of these are currently part of the settings screen. Let's leave them out for now so if/when they are added, whoever adds them has to go through the process of modifying this section of code.